### ISSUE TITLE
How to check huey task results in django test cases?

### ISSUE THREAD

**Issue by elpaso** [coleifer/huey#593]
Hi, my apologies if this is not the right place to ask: this is probably me not understanding the API.

My use case is a django API where the first call will trigger the task and the caller gets back the task id.
Another API will take the task id and give back the task status (and possibly its progress) and the task results if the task has completed (or an error code otherwise).

The question is how to retrieve the result of the task in the test case after the first API call has returned. 

I understand that in immediate mode the call is executed immediately, but I cannot find the results anywhere in the HUEY object.

What is the recommended way to test the correct behavior of the asynchronous API call and check the result?

**Comment by coleifer** [comment]
By default huey will run in "immediate"-mode when Django DEBUG is set to True. The thinking is that when you are testing or developing your huey application, you don't want to run a separate consumer process. Immediate mode executes the task code synchronously and returns a `Result` handle with the task result ready-to-go. Immediate-mode uses an in-memory storage, so the result should be stored there.

Simple example:

```python
from huey import *

huey = MemoryHuey(immediate=True)

@huey.task()
def add(a, b):
    return a + b

res = add(1, 2)
print(res)
print(res.get())
```

Output:

```
<Result: task ced0d8c0-1345-45ca-a89f-8c7ca3030344>
3
```

In other words, everything should be the same, just synchronous.
